include/deal.II/cgal/utilities.h

```diff
@@ -0,0 +1,110 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_cgal_utilities_h
+#define dealii_cgal_utilities_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_CGAL
+#  include <CGAL/Cartesian.h>
+#  include <CGAL/Simple_cartesian.h>
+
+DEAL_II_NAMESPACE_OPEN
+/**
+ * Interface to the Computational Geometry Algorithm Library (CGAL).
+ *
+ * CGAL is a software project that provides easy access to efficient and
+ * reliable geometric algorithms. The library offers data structures and
+ * algorithms like triangulations, Voronoi diagrams, Boolean operations on
+ * polygons and polyhedra, point set processing, arrangements of curves, surface
+ * and volume mesh generation, geometry processing,  alpha shapes, convex hull
+ * algorithms, shape reconstruction, AABB and KD trees...
+ *
+ * You can learn more about the CGAL library at https://www.cgal.org/
+ */
+namespace CGALWrappers
+{
+  /**
+   * Convert from deal.II Point to any compatible CGAL point.
+   *
+   * @tparam CGALPointType Any of the CGAL point types
+   * @tparam dim Dimension of the point
+   * @param [in] p An input deal.II Point<dim>
+   * @return CGALPointType A CGAL point
+   */
+  template <typename CGALPointType, int dim>
+  inline CGALPointType
+  to_cgal(const dealii::Point<dim> &p);
```

Comment:
Do we have any precedence in such a case? According to

https://github.com/dealii/dealii/blob/f875ac3bce89586c5a32eb74b4b77fc0e9a3d26c/include/deal.II/grid/reference_cell.h#L537-L538

The right name here would be `deal(ii)_point_to_cgal_point`.

Reply:
The function above refers to index ordering, not to points. That function should have been called then `exodusii_vertex_index_to_dealii_vertex_index` if we wanto to go down that route... :)

Reply:
We have a precedence: 

https://www.dealii.org/current/doxygen/deal.II/namespaceOpenCASCADE.html#a7b0b2393711bae58d859be1bb9761781

the function is called `point` there, in both cases. We could rename this as 

~~~
auto cgal_p = CGALWrappers::point<CGAL::Point_3<Kernel>>(p);
auto dealiip = CGALWrappers::point<3>(cgal_p);
~~~

Reply:
@peterrum, what do you want me to do? 

1. `to_dealii_point` and `to_cgal_point`
2. `point` in both cases
3. `dealii_point_to_cgal_point` and `cgal_point_to_dealii_point`
4. leave it as it is now

Reply:
My preference would be to be verbose:

> dealii_point_to_cgal_point and cgal_point_to_dealii_point

But I would be fine with `to_dealii_point`. Do others have an opionion?

Reply:
If nobody replies by tonight, we'll rename them to `dealii_point_to_cgal_point` and `cgal_point_to_dealii_point`

Reply:
> If nobody replies by tonight, we'll rename them to dealii_point_to_cgal_point and cgal_point_to_dealii_point

... and I'll merge it so that the other PRs are not blocked!

Reply:
This should be ok now.